parlai/core/agents.py

```diff
@@ -189,6 +189,18 @@ def __init__(self, opt, shared=None):
         self.task_idx = -1
         self.new_task = True
         self.random = opt.get('datatype') == 'train'
+        # Make multi-task task probabilities.
+        self.cum_task_weights = [1] * len(self.tasks)
+        self.task_choices = range(len(self.tasks))
+        weights = self.opt.get('multitask_weights', [1])
+        sum = 0
+        for i, w in enumerate(self.tasks):
```

Comment:
you don't seem to use `w` in this loop so you can probably just iterate through `self.task_choices`

Reply:
fixed